src/meadowrun/aws_integration/ec2.py

```diff
@@ -251,11 +251,21 @@ class LaunchEC2InstanceSuccess(LaunchEC2InstanceResult):
     """
     Indicates that launch_ec2_instance succeeded. public_address_continuation can be
     awaited to get the public address of the launched instance.
+    public_address_continuation can be None which indicates that the launch of the
+    instance was cancelled. If the instance was mid-launch, then the instance was
+    terminated before it could fully launch.
     """
 
-    public_address_continuation: Awaitable[str]
+    public_address_continuation: Awaitable[Optional[str]]
     instance_id: str
 
+    @classmethod
+    def get_cancelled(cls) -> LaunchEC2InstanceSuccess:
+        """This is used to indicate a cancelled launch"""
+        none_future: asyncio.Future[None] = asyncio.Future()
+        none_future.set_result(None)
+        return LaunchEC2InstanceSuccess(none_future, "")
```

Comment:
Kinda weird. Should `public_address_continuation` hold a `Task` instead, which can then just be cancelled as usual?

Reply:
Okay, my thought was that Task cancellations just show up as `asyncio.CancelledError`, and we can't distinguish between Ctrl+C vs an internally initiated cancellation (I think)? E.g. in `launch_ec2_instances`, if we have 2 instances that launched successfully and 1 that was cancelled, if this was an internally initiated cancellation then we should just return the 2 that were launched successfully. Whereas if it's a Ctrl+C then we need to re-raise the asyncio.CancelledError

Reply:
Partially addressed as per conversation by adding a LaunchEC2InstanceAborted result type